docs/execution/ExecutionTwists.md

```diff
@@ -95,3 +95,43 @@ Once all jobs have been reconnected to, the workflow will keep running normally.
 
 During the reconnection process Cromwell might ask backends to reconnect to jobs that were never started before the restart. In that case, the job will be mark as failed with an explanation message. This failure is benign and only an artifact of the fact that Cromwell was restarted.  
 If the backend does not support reconnection to an existing job, jobs will be marked as failed with an explanation message as well. The backend status of the jobs will be "Unknown".
+
+## Graceful Shutdown
+
+When Cromwell is run as a server, it will by default attempt to gracefully shutdown, stopping its different services in a specific order to avoid losing critical data.
+This behavior, documented below, can be turned off in the configuration via `system.graceful-server-shutdown = false`.
+
+Upon receiving a `SIGINT` or `SIGTERM` signal, the JVM will initiate its shutdown process. Prior to this Cromwell will attempt to shutdown its own services in the following way:
+
+1. Workflows in `Submitted` state are no longer started
+2. Cromwell unbinds from the address/port it was listening on. From this point the Cromwell server is unreachable via the endpoints.
+3. All actors generating data that needs to be persisted receive a message asking them to gracefully stop.
+This means that they are given some time (see below for how much and how to change it) to return to a known "consistent" state.
+For example, an actor waiting for a response from the database before sending information to the metadata will wait for that response before shutting itself down.
+4. All active connections from the REST endpoints are completed and closed. At this point any client that made a request before the shutdown process started should have received a response.
+5. All actors responsible for data persistence are in turn being asked to gracefully shutdown. 
+For example, all queued up metadata writes are executed.
+6. Database connection pools are shutdown.
+7. Actor system shuts down.
+8. JVM exits.
+    
+This multi-stage process is designed to minimize the risk of data loss during shutdown. However in order to prevent this process to last forever, each stage (called phase) has its own timeout.
+If the phase does not complete within the given timeout, actors will be forcefully stopped and the next phase will start.
+
+This logic is implemented using [Akka Coordinated Shutdown Extension](http://doc.akka.io/docs/akka/current/scala/actors.html#coordinated-shutdown).
```

Comment:
ToL: I never know what to do about linking to external-docs combined with the fact that we _rarely_ update our-docs when we update our libraries. The link here is to Akka's "[current](http://doc.akka.io/docs/akka/current/scala/actors.html#coordinated-shutdown)" floating docs URL, while we're currently [pinned](https://github.com/broadinstitute/cromwell/blob/631ec41c6303321d0d5d384061f742543782a746/project/Dependencies.scala#L5) the akka library version "2.5.4" which is close but not technically the current version. We could change our doc link for now to point to [this](https://doc.akka.io/docs/akka/2.5.4/scala/actors.html#coordinated-shutdown) permalink instead. But if someone updates the library I'm not betting they will come back and fix the permalink link too…

Reply:
That's a good point. I think I'll link to both the "current" and the "2.5.4" docs, so that if we do happen to forget to update the docs (which is impossible, I know), that users can still find the latest docs.